Develop1

pValevach opened the pull request:
Подключил библиотеки: Jacoco, Mockito, JUnit 4.
Создал тестовые классы для Bun, Burger, Ingredient, IngredientType , а также отчёт в Jacoco.

Commit:
Добавлены тесты и отчет в jacoco

Commit:
Updating pom.xml

mkouznetsov-git commented on src/test/java/BunTest.java:
```diff
@@ -0,0 +1,33 @@
+
+    @Before
+    public void setup() {
+        bun = new Bun("Булочка", 10);
```
Хорошо бы проверить различные варианты задания названия для булки. Длинное название, короткое, пустая строка, null и т.п.

mkouznetsov-git commented on src/test/java/BunTest.java:
```diff
@@ -0,0 +1,33 @@
+        float expectedPrice = 10;
+        float actualPrice = bun.getPrice();
+
+        Assert.assertEquals(expectedPrice, actualPrice, 0);
```
Слово Assert здесь лишнее, т.к. ты уже сделал статический импорт assertEquals.

mkouznetsov-git commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,127 @@
+
+    @Test
+    public void setBunTest() {
+        burger.setBuns(bun);
```
Здесь используется пустая булка bun, которая не была инициализирована. Проверка некорректна.

mkouznetsov-git commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,127 @@
+                "= filling Говяжья котлетка =\r\n" +
+                "(==== Булочка с корицей ====)\r\n" +
+                "\r\n" +
+                "Price: 300,000000\r\n";
```
На некоторых системах разделителем дробной части является запятая, а на некоторых (как у меня, например, на маке) -- точка. И это приводит к падению теста. Лучше всего использовать форматирование ожидаемого значения, а не писать "300,000000" сразу в строку.

Commit:
1. Добавлена параметризация в BunTest.java.
2. Во всех тестовых классах прямой импорт метода .assertEquals.
3. Инициализирован объект bun в тесте setBunTest, тестовый класс BurgerTest.java.
4. Изменено поле expectedReceipt в тесте getReceiptTest, тестовый класс BurgerTest.java.
